### PR TITLE
EL 6.0 new tests

### PR DESCRIPTION
**Pull request by markt-asf**
**Fixes Issue**
New tests for functionality added in EL 6.0

**Related Issue(s)**
None.

**Describe the change**
- `ArrayELResiolver` tests extended to include new `length` property
- Add tests for new `RecordELResolver`
- Add tests for new `OptionalELResolver`
- Add test for method visibility clarification
There are also some clean-up commits (no functional changes) to the touched classes to fix various IDE warnings.

**Additional context**
None.

CC @alwin-joseph @anajosep @arjantijms @cesarhernandezgt @dblevins @m0mus @edbratt @gurunrao @jansupol @jgallimore @kazumura @kwsutter @LanceAndersen @bhatpmk @RohitKumarJain @shighbar @gthoman @brideck @OndroMih @dmatej
@starksm64 @scottmarlow

### CODE REVIEW

File: el/src/main/java/com/sun/ts/tests/el/api/jakarta_el/arrayelresolver/ELClientIT.java [hunk 1]
```diff
@@ -310,4 +308,34 @@ public void arrayELResolverCCETest() throws Exception {
...
+    boolean pass;
+    StringBuffer buf = new StringBuffer();
+    String[] colors = { "red", "blue", "green" };
+
+    try {
+      ArrayELResolver arrayResolver = new ArrayELResolver();
+      BareBonesELContext barebonesContext = new BareBonesELContext();
+      ELContext context = barebonesContext.getELContext();
+
+      pass = ResolverTest.testELResolver(context, arrayResolver, colors,
```
Comment (alwin-joseph) [inline el/src/main/java/com/sun/ts/tests/el/api/jakarta_el/arrayelresolver/ELClientIT.java:330]: I get below exception when run with Glassfish8

```
java.lang.Exception: java.lang.NumberFormatException: For input string: "length"
        at com.sun.ts.tests.el.api.jakarta_el.arrayelresolver.ELClientIT.arrayELResolverLengthTest(ELClientIT.java:333)
Caused by: java.lang.NumberFormatException: For input string: "length"
        at com.sun.ts.tests.el.api.jakarta_el.arrayelresolver.ELClientIT.arrayELResolverLengthTest(ELClientIT.java:330)

```

**Comment by markt-asf** [comment]
Glassfish may need an EL API update. I'm working on a 6.0.0-M2 now.

**Review by alwin-joseph — APPROVED** [review]
All tests except arrayELResolverLengthTest pass with Glassfish8-M1. Will retest when EL api M2 is integrated with glassfish.

**Comment by markt-asf** [comment]
EL 6.0.0-M2 is now available on Maven Central

**Comment by alwin-joseph** [comment]
> EL 6.0.0-M2 is now available on Maven Central

I was trying EL 6.0.0-M2 with Glassfish. arrayELResolverLengthTest passed.
But the test beanELResolverMethodVisibilityTest fails with below exception. 

```

[ERROR] com.sun.ts.tests.el.api.jakarta_el.beanelresolver.ELClientIT.beanELResolverMethodVisibilityTest  Time elapsed: 0.027 s  <<< ERROR!
java.lang.Exception: 
Test FAILEDsetValue() tested successfully on writable property
Invalid value from getValue():
Value expected: 0
Value retrieved: 0
getType() returns int
non-null as expected
isReadOnly() returns false as expected
getCommonPropertyType() returns java.lang.Object

        at com.sun.ts.tests.el.api.jakarta_el.beanelresolver.ELClientIT.beanELResolverMethodVisibilityTest(ELClientIT.java:419)
```

**Comment by markt-asf** [comment]
I'll look into that. It was passing for me locally with Tomcat.

**Comment by alwin-joseph** [comment]
> I'll look into that. It was passing for me locally with Tomcat.

It turns out the test passed when the java version is 21, fails in JDK 17.

**Comment by alwin-joseph** [comment]
> Thanks. That is a useful data point. I will look at this but if it is a Java 17 / Java 21 issue then that puts it a lot lower down my priority list.

All tests passed against Glassfish with latest commits. Thanks.

**Pull request merged**